src/mRides-server/Controllers/UserController.cs

```diff
@@ -98,14 +98,36 @@ public long getGSD(int id)
         /// </summary>
         // POST api/values
         [HttpPost]
-        public long setGSD([FromHeader]string id, [FromBody]dynamic sentObject)
+        public long setGSD([FromBody]dynamic sentObject)
         {
-            int newUserId = Convert.ToInt32(id);
-            //long newAmountGSD = long.Parse(sentObject.amountGSD);
+            int newUserId = sentObject.userId;
             int newAmountGSD = sentObject.amountGSD;
             _userCatalog.setGSD(newUserId, newAmountGSD);
             return _userCatalog.get(newUserId).GSD;
         }
-       
+
+        /// <summary>
+        /// Returns the gender of the user with the corresponding id
+        /// </summary>
+        [HttpGet("{id}")]
+        public string getGender(int id)
```

Comment:
Since we get the whole user object(which already has this attribute) to populate the user profile, this method is redundant (YAGNI)